src/charm.py

```diff
@@ -106,6 +114,32 @@ def on_config_changed_handler(event, fw_adapter):
         time.sleep(1)
 
 
+def on_new_am_rel_handler(event, fw_adapter):
+    if not fw_adapter.am_i_leader():
+        logging.debug("Unit is not a leader, skip pod spec configuration")
+        return
+
+    alerting_config = json.loads(event.data.get('alerting_config', '{}'))
+    logger.debug(
+        "Got alerting config: {} {}".format(type(alerting_config),
+                                            alerting_config)
+    )
+
+    logging.debug("Building Juju pod spec with alerting config")
```

Comment:
Why can't we add this logic to the `set_juju_pod_spec` in `src/charm.py`?

Reply:
Yeah, I should. Heh.